SECURITY.md

```diff
@@ -0,0 +1,57 @@
+# Extensions Security Guidelines
+
+OpenSearch's support for extensions allows for taking already powerful use cases and expanding on them, this creates a larger surface area for misuse, vunerabilities, and malicious interactions.  This document outlines several areas for enhancements, features, and practices to incorperate into extensions for OpenSearch
+
+To keep concepts consistant, this document is using terminology from [NIST Glossary](https://csrc.nist.gov/glossary).
+
+Additional terms:
+* **Plugin** - reference to the existing functionality to extend OpenSearch functionality.
+* **Extension** - reference to the in development functionality to extend OpenSearch.
+
+## Host security
+
+The Java Security Manager is the mechanism for ensuring plugins are limited in what they can do to the host operation system resources (cpu/disk/memory/network/...).  As there are limitations and its deprecated with removal scheduled in the next release of the JVM.
+
+The current extensions design they operate via Rest APIs, by isolating extensions from using host they are prevented from executing operation system calls directly on hosts of the cluster.
+
+## Communications security (COMSEC)
+
+Data is transferred from the OpenSearch cluster to the extensions.  This is done via https requests between the nodes on the cluster and the extensions endpoint(s).
+
+Extensions should never directly communicate with other extensions, cross extensions work should always be proxied through OpenSearch.
+
+## Data Security
+
+OpenSearch stores data in its memory and local file system storage, the security plugin provides mechanisms to control data access within OpenSearch.  Extensions have independent data storage.
+
+Plugins store data inside of the OpenSearch cluster itself such as in system/hidden indices.
+
+## Access Control
+
+OpenSearch offers access control through the security plugin, with checks action names and filters.  Actions registered within OpenSearch that are not permitted never reach the handler for a plugin or extension execution.
+
+Resource level access control is governed by the extension, when requests are processed the [user](https://github.com/opensearch-project/common-utils/blob/main/src/main/java/org/opensearch/commons/authuser/User.java) object from common-utils is checked for matching backendroles/roles.  Access control checks are managed wholy in the plugin.  Example from anomaly detection, [checkUserPermissions](https://github.com/opensearch-project/anomaly-detection/blob/875b03c1c7596cb34d74fea285c28d949cfb0d19/src/main/java/org/opensearch/ad/util/ParseUtils.java#L568).
```

Comment:
This needs an "As is" and "To Be" summary.  We are planning to do something else with User, aren't we?  You're also mixing extension and plugin here, not sure if this is documenting current vs. future.   I would like to put RBAC as a goal.

Reply:
> I would like to put RBAC as a goal.

Me too, I think there is a version of this document that includes lists of issues to be implemented/decisions to make.  Before we come to the possible implementations I'd like to lay out the surface area for prioritization.  How would you feel about holding off on individual features until we codify this framework?

Reply:
As mentioned in earlier comment, we do need an individual feature list somewhere. We can limit this particular document to the overall framework/scope but identifying at least major categories of features is needed for any actual planning and scoping of future work

Reply:
Lets revisit this in a follow up PR to make sure we've got the detail we need